### ISSUE TITLE
Linux support

### ISSUE THREAD

**Issue by MGThePro** [antonpup/Aurora#640]
I'm not sure if this is possible at this point, since most of the code is in C#, but I would enjoy to see Linux support in Aurora.

**Comment by brainbug89** [comment]
Most SDKs are also written for Windows. So even if Aurora would work on Linux... there aren't many supported lighting devices.

**Comment by simon-wh** [comment]
Yeah, as @brainbug89 said the Lighting SDKs are only really available for Windows and to write a custom one would require access to all the devices unless you could find the details of the HID interface. Added to that the UI is in WPF which is Windows only as well, but that is beside the point. If the SDK's were available for Linux as well, I'd be willing to do the work to support it but it isn't really an option.

**Comment by simon-wh** [comment]
Just out of curiosity, for anyone who ends up on this issue and is interested in the possibility for Linux support, what would the demand be like? I would be tempted to work on a Linux version but it wouldn't have the same set of available games and devices at the very least. I haven't looked into alternative apis for the windows only ones in use such as screen capture for ambilight, audio capture for visualiser layer, input capture for interactive layer etc.

**Comment by diogotr7** [comment]
I think Linux support is a good thing, but in a program such as this one, without official support from many manufacturers, I unfortunately don't think that it would be very useful.

I really wish major companies supported linux at least as much as they support mac, but I really don't see that happening.

**Comment by simon-wh** [comment]
@diogotr7 Well, at least for device support there are quite a lot of open source linux libs that interact with a lot of the supported devices like for [Logitech](https://github.com/MatMoul/g810-led), [Razer](https://github.com/openrazer/openrazer), [Corsair](https://github.com/kevhawk2914/ckb) are the ones that cover a lot of the devices. There are bits and pieces here and there for other brands. The main problem with supporting devices without sdk's is actually having the devices to reverse engineer them, but we don't need to worry about that if it's already been done for us.

For games it would prob be really weak as we could definitely get Dota and CSGO, but other games that are achieved through the wrappers most likely would have no chance of working on linux, so it'd mainly be desktop lighting and the limited games that would work.

**Comment by Freso** [comment]
Roccat's devices have [open source utilities](http://roccat.sourceforge.net/) that interface with them, incl. colour changing. [Wooting](https://www.wooting.nl/) is trying to open source as much as they can and are actively working on providing full support for their keyboards under Linux. I used to have a Razer keyboard, IIRC that also had some open source utilities, but I can't remember to which extent they worked.

I haven't looked into exactly what [libgaminggear](https://sourceforge.net/projects/libgaminggear/) provides/aims to do, but that might also be something to consider working with.

**Comment by filipegmedeiros** [comment]
Nothing yet?

**Comment by simon-wh** [comment]
@filipegmedeiros I'm currently planning out the rewrite which will have Linux support

**Comment by filipegmedeiros** [comment]
@simon-wh  it's my dreams see to make my Desktop Settings and Dota 2 on Arch Linux. I use windows just really Just, because aurora. :(

**Comment by diogotr7** [comment]
It's being worked on in the dev-rewrite branch @filipegmedeiros

**Comment by mirh** [comment]
> but other games that are achieved through the wrappers most likely would have no chance of working on linux

Why not?
I can think of plenty of apis (from vulkan, to openal, to asio) that can be passed-through wine.

**Comment by simon-wh** [comment]
@mirh Yeah, I dunno why I said that, should work with stuff that runs under wine/dxvk.

**Comment by mirh** [comment]
If any the actual wonder I have, is how many of those games that got a native linux port (especially if they used special windows-only sdks like payday I guess?), still have the right functions to hook available there.

**Comment by filipegmedeiros** [comment]
Any news about linux version? xD

**Comment by filipegmedeiros** [comment]
Any news about linux version? xD

**Comment by simon-wh** [comment]
@filipegmedeiros did you just delete your comment and repost it because I didn't respond? hahahaha, you sneaky bugger

If you insist, it's being worked on with the rewrite, Linux support is going to be there intrinsically, as I only use Linux. It's coming along a lot slower than I originally intended as I've been struggling to find time for Uni work and the rewrite, but hopefully it's not super far away. Once I have anything substantial you'll likely see it in the Aurora discord server.

**Comment by N00byKing** [comment]
Hi, just wanted to ask: Will the Linux version integrate with wine or run native ?

**Comment by drohm** [comment]
I look forward to testing this as soon as it gets to a usable state.

**Comment by HeroCC** [comment]
Hey Simon! Any update on this? I know for at least Corsair there's https://github.com/ckb-next/ckb-next which works in Linux. I don't see much work on the dev-rewrite branch, if there's anything you need help with please let us know! I can do hardware testing (I'm a software dev but I haven't done much C# work).

**Comment by gardotd426** [comment]
I would love to know if there's any progress on this

**Comment by NicTanghe** [comment]
For corsair on linux you can use the following for inspiration.
https://github.com/ckb-next/ckb-next

**Comment by NicTanghe** [comment]
In regard to your question.
The most important features for me in order would be Time, cpu stats, Ram stats, gpu stats, audiovis, ambilight.
Those probably aren't hard to implement at all since these are just text documents somewhere being updated.

Having my time available to me in binary on my keyboard is a godsend feature.
Hardware stats are also very useful.

**Comment by NicTanghe** [comment]
2 years since last commit can we assume linux version has been abbandoned ?

**Comment by terencode** [comment]
It might not do everything Aurora does, but there is https://gitlab.com/CalcProgrammer1/OpenRGB.

**Comment by CalcProgrammer1** [comment]
Now that the Aurora OpenRGB backend is mostly functional and OpenRGB has been updated to have consistent key names, I think it would be a good idea to start looking at this again.  It looks like Linux support was considered but not worked because no SDKs existed on Linux.  With Aurora now having an OpenRGB backend, this is no longer true.  The OpenRGB backend should allow Aurora integration with Razer, Corsair, HyperX, Logitech, and more keyboard brands on Linux.

For status of OpenRGB/Aurora integration work see this ticket: https://github.com/antonpup/Aurora/issues/1962

**Comment by BRA1L0R** [comment]
ETA?

**Comment by NicTanghe** [comment]
> 
> 
> ETA?

I fear the devs stopped using linux.

**Comment by BRA1L0R** [comment]
> > ETA?
> 
> I fear the devs stopped using linux.

@NicTanghe that would be unfortunate

**Comment by JoeJoeTV** [comment]
Is this still going to happen?
I love the project, but I am in the process of switching to linux mostly and I have also seen OpenRGB, which looks quite promising.

**Comment by NicTanghe** [comment]
I think he just added this to milestone 1.0.0 i might be mistaken though (whould be awsome)
EDIT: It was probably just your message that pinged me.

**Comment by CalcProgrammer1** [comment]
> 
> 
> > ETA?
> 
> I fear the devs stopped using linux.

I think the bigger fear is that there are no more devs.  The Aurora Discord server is mostly just questions, and the consensus there is that Aurora is essentially a dead project.  There are a few devs who do small maintenance things, and one who is making custom builds to add more OpenRGB device support, but the idea I get from there is that Aurora is a mess of spaghetti code that no one wants to seriously maintain anymore, certainly not to the extent of replacing the UI framework with something that is Linux compatible.

The developer who added OpenRGB support, @diogotr7, has been mostly focusing on the Artemis project.  Artemis does work on Wine, though it has some graphical glitches.  He wrote a plugin for OpenRGB support in Artemis.  It looks like the current goal is to port effects from Aurora into Artemis plugins and then transition to using Artemis instead.

**Comment by serious-angel** [comment]
> Artemis does work on Wine...

Is there any issue running a .NET 5 app on Linux natively?

**Comment by CalcProgrammer1** [comment]
I'm not a C# developer, but from what I understand the UI framework is Windows only, so even though Aurora and Artemis are C#, they won't be able to build/run natively on Linux.  There are alternative UI frameworks that are cross platform, but the code would have to be modified to use them.

**Comment by serious-angel** [comment]
> I'm not a C# developer, but from what I understand the UI framework is Windows only, so even though Aurora and Artemis are C#, they won't be able to build/run natively on Linux. There are alternative UI frameworks that are cross platform, but the code would have to be modified to use them.

> .NET 5.0 has a nearly identical platform support matrix as .NET Core 3.1, for Windows, macOS, and Linux. If you are using .NET Core 3.1 on a supported operating system, you should be able to adopt .NET 5.0 on that same operating system version for the most part.
> Source: https://devblogs.microsoft.com/dotnet/announcing-net-5-0/

In other words, the `.NET 5` supports Windows and Linux natively(~ .NET Framework + .NET Core).
Oh, you meant the UI. Indeed, the WPF is not currently supported on Linux, but there is https://github.com/AvaloniaUI/Avalonia, for example.

**Comment by JoeJoeTV** [comment]
> > > ETA?
> > 
> > 
> > I fear the devs stopped using linux.
> 
> I think the bigger fear is that there are no more devs. The Aurora Discord server is mostly just questions, and the consensus there is that Aurora is essentially a dead project. There are a few devs who do small maintenance things, and one who is making custom builds to add more OpenRGB device support, but the idea I get from there is that Aurora is a mess of spaghetti code that no one wants to seriously maintain anymore, certainly not to the extent of replacing the UI framework with something that is Linux compatible.
> 
> The developer who added OpenRGB support, @diogotr7, has been mostly focusing on the Artemis project. Artemis does work on Wine, though it has some graphical glitches. He wrote a plugin for OpenRGB support in Artemis. It looks like the current goal is to port effects from Aurora into Artemis plugins and then transition to using Artemis instead.

How can I use Artemis on Wine and is there hope that it will come to linux or other platforms?

**Comment by NicTanghe** [comment]
people should just stop using windowsshit and use qt instead

**Comment by EXtremeExploit** [comment]
so all we need is some way to translate Aurora profiles and config to Atermis? would game integrations even work?
